sleuth_backend/solr/models.py

```diff
@@ -0,0 +1,73 @@
+import json
+import time
+from . import connection as solr
+
+class SolrDocument(object):
+    """
+    An base class for documents that are inserted into Solr, an retured as 
+    search results from the Sleuth API.
+    """
+
+    def __init__(self, doc, **kwargs):
+        """
+        This method should be called by the subclass constructor.
+        """
+        self.doc = doc
+        for key in self.doc.keys():
+            if key in kwargs and key is not "type":
+                doc[key] = kwargs[key]
+
+    def output_format(self):
+        """
+        Returns the document in JSON format as it is supposed to be returned 
+        from a Sleuth API response. This method must be overridden.
+        """
+        pass
+
+    def save_to_solr(self, solr_connection):
+        """
+        Submits the document to the given Solr connection. This method should not
+        be overridden.
+        """
+        solr_connection.insert_document(self.type(), self.doc)
+
+    def type(self):
+        """
+        Returns a string representing the document type. This method should not
+        be overridden.
+        """
+        return self.doc["type"]
+
+class GenericPage(SolrDocument):
+    """
+    Represents a generic web page.
+    """
+    doc = {
+        "id": "",
+        "type": "genericPage",
+        "siteName": "",
+        "updatedAt": "",
+        "pageName": "",
+        "pageTitle": "",
+        "content": ""
+    }
+
+    def __init__(self, **kwargs):
+        super(GenericPage, self).__init__(self.doc, **kwargs)
+        if 'blurb' in kwargs:
+            self.blurb = kwargs['blurb'] 
+        else:
+            self.blurb = ""
```

Comment:
Why is `blurb` defined here and not:
* upon extraction (search-relevant extract from `content`) or
* upon insertion (to keep in Solr)?

Reply:
Oh nevermind I remember how this works now

Reply:
@bfbachmann I think it would probably work best if blurb is dynamically pulled from `content` in our API call, instead of as part of the schema